### PR TITLE
Created scripts/hhs_doi_cli.py and updated dependencies in pyproject.…

### PR DESCRIPTION
**Pull request by joshlawrimore**
…toml.

@agt24 modified the filter_cli.py CLI to include DOI extraction with pdf2doi. I further modified it and created this PR. @leej3 , If it doesn't break anything, please approve the review. I'm happy to take any suggestions.

### CODE REVIEW

File: scripts/hhs_doi_cli.py [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+
+    Raises
+    ------
+    FileExistsError
+        If `new_run` is True and the output CSV file already exists.
+    """
+    if pdf2doi_verbose:
+        pdf2doi_config.set("verbose", True)
+    else:
+        pdf2doi_config.set("verbose", False)
```
Comment (quang-ng) [inline scripts/hhs_doi_cli.py]:
```suggestion
    pdf2doi_config.set("verbose", pdf2doi_verbose)
```

File: scripts/hhs_doi_cli.py [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+
+    output_exists: bool = output_csv.exists()
+
+    if output_exists and new_run is False:
+        print("Removing previously processed PDFs")
```
Comment (quang-ng) [inline scripts/hhs_doi_cli.py]: could we using logger instead of print?

File: scripts/hhs_doi_cli.py [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+            + " Please remove existing file prior to a new run!"
+        )
+    else:
+        # keep a generator for memory management
+        pdfs = pdf_dir.rglob("*.[pP][dD][fF]")
+        if make_filelist:
+            pdfs = list(pdfs)
+
+    dicts: list[dict] = []
+    with open(output_csv, "a") as f_out:
```
Comment (quang-ng) [inline scripts/hhs_doi_cli.py]: this is too complex, could we break it into smaller functions

File: scripts/hhs_doi_cli.py [hunk 4]
```diff
@@ -0,0 +1,276 @@
...
+                try:
+                    doi_results = pdf2doi.pdf2doi(str(pdf.absolute()))
+                    # Get the first result if multiple are returned
+                    doi_info = (
+                        doi_results[0] if isinstance(doi_results, list) else doi_results
+                    )
+                    identifier = doi_info.get("identifier")
+                    identifier_type = doi_info.get("identifier_type")
+                    extraction_method = doi_info.get("method")
+                except Exception:
```
Comment (quang-ng) [inline scripts/hhs_doi_cli.py]: Insert a logger here so we can identify which exception occurs.

**Review by quang-ng — COMMENTED** [review]
Can we relocate the processing code to the dsst-etl folder and create a unit test for it? Here, we only call the function.

**Review by leej3 — APPROVED** [review]
I haven't run the code but it looks reasonable to merge. I'm assuming this functionality will be reviewed/used in upcoming development.

I agree with @quang-ngs points:
- Using different logging levels will make things easier to debug in the long run.
- Files in the script directory should have little but a CLI... other functionality should be in the package and ideally it should be tested.
- The function is too large which will harm maintainability. The deep nesting can be avoided by turning the core functionality into its own function.

File: scripts/hhs_doi_cli.py (hunk 2 above)
Comment (joshlawrimore) [reply scripts/hhs_doi_cli.py]: Done! Set up a package-level logger at dsst_etl.logger

File: scripts/hhs_doi_cli.py (hunk 4 above)
Comment (joshlawrimore) [reply scripts/hhs_doi_cli.py]: Done!

File: scripts/hhs_doi_cli.py (hunk 3 above)
Comment (joshlawrimore) [reply scripts/hhs_doi_cli.py]: I agree. I broke the main function into 3 and then put all functionout outside of argparse in dsst_etl.hhs_doi.py

**Review by quang-ng — APPROVED** [review]
Great!!! 🚀 🚀 🚀 🚀 🚀 🚀 🚀

**Pull request merged**